src/use_file.rs

```diff
@@ -58,10 +60,13 @@ fn get_rng_fd() -> Result<libc::c_int, Error> {
         #[cfg(any(target_os = "android", target_os = "linux"))]
         wait_until_rng_ready()?;
 
-        let fd = open_readonly(FILE_PATH)?;
+        let file = fs::File::open(FILE_PATH).map_err(map_io_error)?;
+
+        let fd = file.into_raw_fd();
         // The fd always fits in a usize without conflicting with FD_UNINIT.
-        debug_assert!(fd >= 0 && (fd as usize) < FD_UNINIT);
-        FD.store(fd as usize, Relaxed);
+        debug_assert!(fd >= 0);
+        const _: () = assert!(FD_UNINIT < 0);
+        FD.store(fd, Relaxed);
```

Comment:
It isn't strictly guaranteed that this `FD.store()` is going to happen before the mutex is unlocked. It seems like here we'd at least benefit from a compiler fence, something like this:
```
 // Force `FD` to be writen before we release the lock.
            // `Mutex` doesn't guarantee it will wait for the above store
            // before unlocking, because `FD` isn't within the Mutex.
            compiler_fence(Ordering::Release);
```
Although, again, it isn't clear that `Mutex<()>` actually is guaranteed to do anything meaningful.